@xen-orchestra/xapi/xoData.js

```diff
@@ -0,0 +1,78 @@
+'use strict'
+
+const assert = require('node:assert/strict')
+
+// This module provides a way to associate data with a XAPI object.
+//
+// The data is stored as a JSON object in the `other_config` field of the XAPI object, under the name `xo:${uuid.slice(0, 8})}.
+//
+// The UUID prefix must match the object UUID, because the `other_config` entries are preserved when objects are copied (e.g. VM.copy).
```

Comment:
Won't this lead to duplicate entries or old entries from the original VM that will never be cleared?

Reply:
Indeed, but IMHO, it's still better than thinking not be able to know whether the data applies to the current entry or not.

I'm open to better ideas :slightly_smiling_face: